lcm/lcm.go

```diff
@@ -0,0 +1,28 @@
+package lcm
+
+import v1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+
+// This package container interface of harbor cluster service lifecycle manage.
+
+type Controller interface {
+	// Provision the new dependent service by the related section of cluster spec.
+	provision(spec *v1.HarborCluster) (*CRStatus, error)
```

Comment:
Why define the unexported methods in the interface?

Reply:
The implementation of the interface and the interface definition itself may be located at different packages.

Reply:
I think this interface is always invoked in Reconcile. The implemenets of Reconciler interface are also implement the Controller interface, then we can use methods in Reconcile.